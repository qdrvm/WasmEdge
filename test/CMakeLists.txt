# SPDX-License-Identifier: Apache-2.0

add_subdirectory(ast)
add_subdirectory(evmc)
add_subdirectory(loader)
add_subdirectory(proxy)
add_subdirectory(expected)
