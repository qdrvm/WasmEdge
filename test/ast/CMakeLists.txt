# SPDX-License-Identifier: Apache-2.0

add_executable(ssvmASTTests
  moduleTest.cpp
  sectionTest.cpp
  descriptionTest.cpp
  segmentTest.cpp
  typeTest.cpp
  expressionTest.cpp
  instructionTest.cpp
)

target_link_libraries(ssvmASTTests
  PRIVATE
  utilGoogleTest
  ssvmLoaderFileMgr
  ssvmAST
)