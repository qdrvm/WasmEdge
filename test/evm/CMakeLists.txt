# SPDX-License-Identifier: Apache-2.0

configure_files(
  ${CMAKE_CURRENT_SOURCE_DIR}/ethereumTestData
  ${CMAKE_CURRENT_BINARY_DIR}/ethereumTestData
  COPYONLY
)

add_executable(ssvmEVMTests
  erc20Test.cpp
)

target_link_libraries(ssvmEVMTests
  PRIVATE
  utilGoogleTest
  utilKeccak
  ssvmVM
)
