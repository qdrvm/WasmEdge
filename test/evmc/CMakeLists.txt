# SPDX-License-Identifier: Apache-2.0

add_executable(ssvmEVMCTests
  evmcTest.cpp
  example_host.cpp
)

target_link_libraries(ssvmEVMCTests
  PRIVATE
  utilEVMC
  utilGoogleTest
  ssvmVM
  ${CMAKE_DL_LIBS}
)
