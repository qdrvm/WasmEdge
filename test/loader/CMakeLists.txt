add_executable(ssvmLoaderFileMgrTests
  filemgrTest.cpp
)

add_executable(ssvmLoaderWagonTests
  wagonTest.cpp
)

add_executable(ssvmLoaderEthereumTests
  ethereumTest.cpp
)

configure_files(
  ${CMAKE_CURRENT_SOURCE_DIR}/filemgrTestData
  ${CMAKE_CURRENT_BINARY_DIR}/filemgrTestData
  COPYONLY
)
configure_files(
  ${CMAKE_CURRENT_SOURCE_DIR}/wagonTestData
  ${CMAKE_CURRENT_BINARY_DIR}/wagonTestData
  COPYONLY
)
configure_files(
  ${CMAKE_CURRENT_SOURCE_DIR}/ethereumTestData
  ${CMAKE_CURRENT_BINARY_DIR}/ethereumTestData
  COPYONLY
)

target_link_libraries(ssvmLoaderFileMgrTests
  PRIVATE
  utilGoogleTest
  ssvmLoaderFileMgr
)

target_link_libraries(ssvmLoaderWagonTests
  PRIVATE
  utilGoogleTest
  ssvmLoaderFileMgr
  ssvmASTLoad
)

target_link_libraries(ssvmLoaderEthereumTests
  PRIVATE
  utilGoogleTest
  ssvmLoaderFileMgr
  ssvmASTLoad
)
