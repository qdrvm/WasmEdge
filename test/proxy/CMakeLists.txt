# SPDX-License-Identifier: Apache-2.0
add_executable(ssvmProxyTests
  proxyTest.cpp
)

configure_files(
  ${CMAKE_CURRENT_SOURCE_DIR}/inputJSONTestData
  ${CMAKE_CURRENT_BINARY_DIR}/inputJSONTestData
  COPYONLY
)

configure_files(
  ${CMAKE_CURRENT_SOURCE_DIR}/outputJSONTestData
  ${CMAKE_CURRENT_BINARY_DIR}/outputJSONTestData
  COPYONLY
)

configure_files(
  ${CMAKE_CURRENT_SOURCE_DIR}/bytecodeTestData
  ${CMAKE_CURRENT_BINARY_DIR}/bytecodeTestData
  COPYONLY
)

target_link_libraries(ssvmProxyTests
  PRIVATE
  utilGoogleTest
  utilKeccak
  ssvmVM
  ssvmProxy
)
