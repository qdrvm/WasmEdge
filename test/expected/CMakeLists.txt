# SPDX-License-Identifier: Apache-2.0

add_executable(expectedTests
  assignment.cpp
  bases.cpp
  constexpr.cpp
  constructors.cpp
  emplace.cpp
  extensions.cpp
  gtest.cpp
  issues.cpp
  noexcept.cpp
  observers.cpp
  relops.cpp
  swap.cpp
)

target_link_libraries(expectedTests
  PRIVATE
  utilGoogleTest
)
